text/0249-v2-experiments.md

```diff
@@ -199,6 +199,82 @@ Disadvantages:
   mitigate this downside by keeping the old unstable package around, but that
   leads to duplicated classes.
 
+### Option 5: superset/subset releases
+
+Instead of *splitting* the stable and experimental APIs into two different
+packages, we release two builds of the same source code, where the *experimental*
+build is a superset of the *stable* build.
+
+This can be done automatically, by using jsii to strip out experimental APIs
+from the **public interface** of the stable release. The implementation
+(`.js` files) will be the same in both releases, so stable APIs in
+`aws-cdk-lib` can safely use experimental APIs internally (though not in
+their public API surfaces).
+
+**How does stripping APIs from the public work?**
+
+| TypeScript            | JavaScript | Java, Python, C#, Go            |
+|-----------------------|------------|---------------------------------|
+| Stripped from `.d.ts` | N/A        | No bindings generated by `jsii` |
+
+In TypeScript, the compiler will prevent usage of APIs that are not advertised
+in the `.d.ts` files. Users will be able to cast this away using `as any` though.
+They will also be able to confuse the compiler enough that they will get to pass
+struct properties that are technically `@experimental`.
+
+For JavaScript, an IDE will (hopefully) not autocomplete APIs missing from
```

Comment:
Why the "hopefully" here? Are you saying there's a chance the IDE _can_ autocomplete APIs missing from the `.d.ts` files? If so, how can we solve that?

If we're certain it can't happen, I would remove the "hopefully", it muddles the message here quite a bit.

Reply:
An IDE can use other mechanisms than reading the `.d.ts` files. It can parse the JavaScript code, it can do dynamic evaluations of the code and inspect the objects at runtime.

I don't know which ones use what mechanism.